test/integration/policy_set_test.go

```diff
@@ -0,0 +1,247 @@
+				func() error {
+					var err error
+					rootPlcSet, err = policySetRsrc.Namespace(userNamespace).Get(
+						context.TODO(), testPolicySetName, metav1.GetOptions{},
+					)
+					return err
+				},
+				defaultTimeoutSeconds*2,
+				1,
+			).Should(BeNil())
+
+			Expect(rootPlcSet.Object["status"]).Should(utils.SemanticEqual(yamlPlc.Object["status"]))
+		})
+
+		It("Should add a status entry in policyset for a policy that does not exist", func() {
+			By("Creating " + testPolicySetPatchYaml)
+			output, err := testcommon.OcHub("apply",
+				"-f", testPolicySetPatchYaml,
+				"-n", userNamespace)
+			By("Creating " + testPolicySetPatchYaml + " result is " + output)
+			Expect(err).To(BeNil())
+
+			plcSetRsrc := clientHubDynamic.Resource(testcommon.GvrPolicySet)
+			var plcSet *unstructured.Unstructured
+			Eventually(
+				func() error {
+					var err error
+					plcSet, err = plcSetRsrc.Namespace(userNamespace).Get(
+						context.TODO(), testPolicySetName, metav1.GetOptions{},
+					)
+					return err
+				},
+				defaultTimeoutSeconds*2,
+				1,
+			).Should(BeNil())
+
+			By("Checking the status of policy set")
+			yamlPlc := utils.ParseYaml("../resources/policy_set/statuscheck-2.yaml")
+			By("Updating the checking YAML clusterName to " + clusterNamespace)
+			yamlPlc = testcommon.UpdateClusterName(yamlPlc, clusterNamespace)
+
+			Expect(plcSet.Object["status"]).Should(utils.SemanticEqual(yamlPlc.Object["status"]))
+		})
+
+		It("Should update to compliant if all its child policy violations have been remediated", func() {
+			By("Patching replicated policy statuses - Compliant")
+			rootPlcRsrc := clientHubDynamic.Resource(testcommon.GvrPolicy)
+			var rootPlc *unstructured.Unstructured
+			Eventually(
+				func() error {
+					var err error
+					rootPlc, err = rootPlcRsrc.Namespace(userNamespace).Get(
+						context.TODO(), testPolicyName, metav1.GetOptions{},
+					)
+					return err
+				},
+				defaultTimeoutSeconds*2,
+				1,
+			).Should(BeNil())
+			rootPlc.Object["spec"].(map[string]interface{})["remediationAction"] = "enforce"
+			_, err := clientHubDynamic.Resource(testcommon.GvrPolicy).Namespace(userNamespace).Update(context.TODO(), rootPlc, metav1.UpdateOptions{})
+			Expect(err).To(BeNil())
+
+			By("Checking the status of policy set")
+			yamlPlc := utils.ParseYaml("../resources/policy_set/statuscheck-3.yaml")
+			By("Updating the checking YAML clusterName to " + clusterNamespace)
+			yamlPlc = testcommon.UpdateClusterName(yamlPlc, clusterNamespace)
+
+			plcSetRsrc := clientHubDynamic.Resource(testcommon.GvrPolicySet)
+			var plcSet *unstructured.Unstructured
+			Eventually(
+				func() error {
+					var err error
+					plcSet, err = plcSetRsrc.Namespace(userNamespace).Get(
+						context.TODO(), testPolicySetName, metav1.GetOptions{},
+					)
+					return err
+				},
+				defaultTimeoutSeconds*3,
+				1,
+			).Should(BeNil())
+
+			Expect(plcSet.Object["status"]).Should(utils.SemanticEqual(yamlPlc.Object["status"]))
```

Comment:
Should use `Eventually` because you may have to wait until the status of policyset to match expected value.